conn.go

```diff
@@ -11,12 +11,20 @@ import (
 	"github.com/libp2p/go-libp2p-core/protocol"
 )
 
+// Stream is a subset of the read-only methods on the network.Stream interface
+type Stream interface {
```

Comment:
Why can't we use the `network.Stream` interface?

Reply:
Good point - I changed the code to override `network.Stream` instead of re-implementing each of its methods to call the underlying implementation